src/transformStateWithClones.js

```diff
@@ -65,7 +65,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const newState = [];
+  const obj = {};
+
+  Object.assign(obj, state);
+
+  for (const transform of transforms) {
+    switch (transform.operation) {
+      case 'addProperties':
```

Comment:
Good use of switch case! I suggest to replace iterating letters with more meaningful names as in the previous task.